mobile/status_request_log_test.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/status-im/status-go/multiaccounts/settings"
+	"github.com/status-im/status-go/signal"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+func TestRemoveSensitiveInfo(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{
+			name:     "basic test",
+			input:    `{"username":"user1","password":"secret123","mnemonic":"mnemonic123 xyz"}`,
+			expected: `{"username":"user1","password":"***","mnemonic":"***"}`,
+		},
+		{
+			name:     "uppercase password field",
+			input:    `{"USERNAME":"user1","PASSWORD":"secret123"}`,
+			expected: `{"USERNAME":"user1","PASSWORD":"***"}`,
+		},
+		{
+			name:     "password field with spaces",
+			input:    `{"username":"user1", "password" : "secret123"}`,
+			expected: `{"username":"user1", "password":"***"}`,
+		},
+		{
+			name:     "multiple password fields",
+			input:    `{"password":"secret123","data":{"nested_password":"nested_secret"}}`,
+			expected: `{"password":"***","data":{"nested_password":"***"}}`,
+		},
+		{
+			name:     "no password field",
+			input:    `{"username":"user1","email":"user1@example.com"}`,
+			expected: `{"username":"user1","email":"user1@example.com"}`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := removeSensitiveInfo(tc.input)
+			if result != tc.expected {
+				t.Errorf("Expected: %s, Got: %s", tc.expected, result)
+			}
+		})
+	}
+}
+
+func TestLogAndCall(t *testing.T) {
+	// Enable request logging
+	requestlog.EnableRequestLogging(true)
+
+	// Create a mock logger to capture log output
+	var logOutput string
+	mockLogger := log.New()
+	mockLogger.SetHandler(log.FuncHandler(func(r *log.Record) error {
+		logOutput += r.Msg + fmt.Sprintf("%s", r.Ctx...)
+		return nil
+	}))
+	requestlog.NewRequestLogger().SetHandler(mockLogger.GetHandler())
+
+	// Test case
+	testFunc := func(param string) string {
+		return "test result: " + param
+	}
+	testParam := "test input"
+	expectedResult := "test result: test input"
+
+	// Call logAndCall
+	result := logAndCallString(testFunc, testParam)
+
+	// Check the result
+	if result != expectedResult {
+		t.Errorf("Expected result %s, got %s", expectedResult, result)
+	}
+
+	// Check if the log contains expected information
+	expectedLogParts := []string{getShortFunctionName(testFunc), "params", testParam, "resp", expectedResult}
+	for _, part := range expectedLogParts {
+		if !strings.Contains(logOutput, part) {
+			t.Errorf("Log output doesn't contain expected part: %s", part)
+		}
+	}
+
+}
+
+func TestGetFunctionName(t *testing.T) {
+	fn := getShortFunctionName(initializeApplication)
+	require.Equal(t, "initializeApplication", fn)
+}
+
+type testSignalHandler struct {
+	receivedSignal string
+}
+
+func (t *testSignalHandler) HandleSignal(data string) {
+	t.receivedSignal = data
+}
+
+func TestSetMobileSignalHandler(t *testing.T) {
+	// Setup
+	handler := &testSignalHandler{}
+	SetMobileSignalHandler(handler)
```

Comment:
[Code Smell #6: Cleanup MobileSignalHandler](https://www.notion.so/Code-Smell-6-Cleanup-MobileSignalHandler-67dda79e7e0f4275b105af557d890b8d?pvs=4)
```suggestion
	SetMobileSignalHandler(handler)
	t.Cleanup(signal.ResetMobileSignalHandler)
```

Reply:
🫣why I forgot..